jrt/src/com/yahoo/jrt/Supervisor.java

```diff
@@ -16,37 +16,10 @@
  **/
 public class Supervisor {
 
-    private class AddMethod implements Runnable {
-        private Method method;
-        AddMethod(Method method) {
-            this.method = method;
-        }
-        public void run() {
-            methodMap.put(method.name(), method);
-        }
-    }
-
-    private class RemoveMethod implements Runnable {
-        private String methodName;
-        private Method method = null;
-        RemoveMethod(String methodName) {
-            this.methodName = methodName;
-        }
-        RemoveMethod(Method method) {
-            this.methodName = method.name();
-            this.method = method;
-        }
-        public void run() {
-            Method m = methodMap.remove(methodName);
-            if (method != null && m != method) {
-                methodMap.put(method.name(), method);
-            }
-        }
-    }
-
     private Transport               transport;
     private SessionHandler          sessionHandler = null;
-    private HashMap<String, Method> methodMap      = new HashMap<>();
+    private final Object            methodMapLock = new Object();
+    private volatile HashMap<String, Method> methodMap = new HashMap<>();
```

Comment:
I think that AtomicReference is the preferred way.
What say ye, @bjorncs and @vekterli

Reply:
`AtomicReference` lets us relax memory ordering requirements and also makes it more obvious in the source code where memory barriers are inserted, so that's two major advantages that I'm aware of. So it makes sense to prefer it going forward, unless the added pointer indirection alloc itself adds unwanted overhead.

Reply:
now using AtomicReference